raft-tech/TANF-app
# Kibana Deployment

## Summary of Changes
- Added manifests to deploy Kibana and the Elastic proxy
- Updated Nginx to act as an auth endpoint for Kibana
- Removed elastic_setup container because it is now irrelevant
- Updated deployment configs
Pull request closes #2790 

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose up
cd tdrs-backend && docker-compose up
```

1. Open http://localhost:3000/ and sign in.
2. Navigate to /kibana and verify you get an error from nginx
3. Add BYPASS_KIBANA_AUTH=yes to your .env
4. Navigate to /kibana and verify you can see Kibana and it feels native to frontend
5. Navigate to https://tdp-frontend-raft.app.cloud.gov and perform the previous steps
  - To see Kibana in the deployed environment follow the steps below to give yourself access. Make sure you have already requested access by way of the frontend. If you already have access, the steps below still apply. If you don't have the cf cli tool, let me know and I can perform the steps for you after you have requested access from the frontend.
  1. Login to cloud foundry with the CLI: `cf login --sso`
  2. Ssh into backend: `cf ssh tdp-backend-raft`
  3. Enable python: `/tmp/lifecycle/shell`
  4. Start Django shell: `python manage.py shell_plus`
  5. Get your user: `user = User.objects.all().filter(email="YOUR_LOGIN_EMAIL_HERE")[0]`
  6. Approve your user: `user.account_approval_status = "Approved"`
  7. Get correct group: `group = Group.objects.get(name="OFA System Admin")`
  8. Add yourself to group: `group.user_set.add(user)`
  9. Give user a fake hhs_id: `user.hhs_id = "111111111111"`
  10. Save: `user.save()`
  11. Verify you can see the frontend now and the Kibana tab.

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x]   Kibana deploys to dev environment
- [x]  Routes between Kibana and Elastic are configured
- [x]  Kibana can connect to Elastic
- [x]  User can navigate to Kibana from frontend

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
